assert/assertions.go

```diff
@@ -1936,7 +1939,7 @@ func EventuallyWithT(t TestingT, condition func(collect *CollectT), waitFor time
 				condition(collect)
 			}()
 		case errs := <-ch:
-			if len(errs) == 0 {
+			if errs == nil {
```

Comment:
How about adding private getter (`failed() bool`) and setter (`fail()`) to `CollectT`? There we could properly document the difference between `errs == nil` and `len(errs) == 0` and have it encapsulated in this struct's method.

Reply:
Well, we could even call them `Failed() bool` and `Fail()`, as [`testing.TB`](https://pkg.go.dev/testing#TB) has such methods.

But let's play it safe for now about API exposure, and make them private for now as you suggest.

:+1: for adding `failed() bool` and `fail()`.